From: Josselin Mouette <joss@debian.org>
Date: Sun, 18 Nov 2007 01:30:52 +0100
Subject: Generate correct Requires.private for gdk-2.0.pc

This contains all necessary dependencies for static linking, moving them
out of Requires.

Based on a GTK+ 3 commit by Matthias Clasen.

Origin: upstream, 3.3.6, commit:b558885221afa5cce0abbd42410877d58d8d2d9a
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=665326
---
 configure.ac  | 14 +++++++++-----
 gdk-2.0.pc.in |  1 +
 2 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0294160..3eaa305 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1325,23 +1325,27 @@ LDFLAGS="$saved_ldflags"
 
 # Pull in gio-unix for GDesktopAppInfo usage, see at least gdkapplaunchcontext-x11.c
 if test "x$gdktarget" = "xx11"; then
-  GDK_PACKAGES="$PANGO_PACKAGES gio-unix-2.0 $X_PACKAGES gdk-pixbuf-2.0 cairo-$cairo_backend"
+  GDK_PACKAGES="$PANGO_PACKAGES gdk-pixbuf-2.0"
+  GDK_PRIVATE_PACKAGES="gio-unix-2.0 $X_PACKAGES cairo-$cairo_backend"
 else
-  GDK_PACKAGES="$PANGO_PACKAGES gio-2.0 gdk-pixbuf-2.0 cairo-$cairo_backend"
+  GDK_PACKAGES="$PANGO_PACKAGES gdk-pixbuf-2.0"
+  GDK_PRIVATE_PACKAGES="gio-2.0 cairo-$cairo_backend"
 fi
 
-GDK_DEP_LIBS="$GDK_EXTRA_LIBS `$PKG_CONFIG --libs $GDK_PACKAGES` $MATH_LIB"
-GDK_DEP_CFLAGS="`$PKG_CONFIG --cflags  gthread-2.0 $GDK_PACKAGES` $GDK_EXTRA_CFLAGS"
+GDK_DEP_LIBS="$GDK_EXTRA_LIBS `$PKG_CONFIG --libs $GDK_PACKAGES $GDK_PRIVATE_PACKAGES` $MATH_LIB"
+GDK_DEP_CFLAGS="`$PKG_CONFIG --cflags  gthread-2.0 $GDK_PACKAGES $GDK_PRIVATE_PACKAGES` $GDK_EXTRA_CFLAGS"
 #
 # If we aren't writing explicit dependencies, then don't put the extra libraries we need
 # into the pkg-config files
 #
 if test $enable_explicit_deps != yes ; then
-  GDK_PACKAGES="$PANGO_PACKAGES gdk-pixbuf-2.0"
   GDK_EXTRA_LIBS=
+else
+  GDK_PACKAGES="$GDK_PACKAGES $GDK_PRIVATE_PACKAGES"
 fi
 
 AC_SUBST(GDK_PACKAGES)
+AC_SUBST(GDK_PRIVATE_PACKAGES)
 AC_SUBST(GDK_EXTRA_LIBS)
 AC_SUBST(GDK_EXTRA_CFLAGS)
 AC_SUBST(GDK_DEP_LIBS)
diff --git a/gdk-2.0.pc.in b/gdk-2.0.pc.in
index 5b2a5d8..9acba30 100644
--- a/gdk-2.0.pc.in
+++ b/gdk-2.0.pc.in
@@ -8,5 +8,6 @@ Name: GDK
 Description: GTK+ Drawing Kit (${target} target)
 Version: @VERSION@
 Requires: @GDK_PACKAGES@
+Requires.private: @GDK_PRIVATE_PACKAGES@
 Libs: -L${libdir} -lgdk-${target}-@GTK_API_VERSION@ @GDK_EXTRA_LIBS@
 Cflags: -I${includedir}/gtk-2.0 -I${libdir}/gtk-2.0/include @GDK_EXTRA_CFLAGS@
